doc/source/serve/production.md

```diff
@@ -0,0 +1,651 @@
+
+Let's use this graph as an example:
+
+```python
+# File name: fruit.py
+
+import ray
+from ray import serve
+from ray.serve.drivers import DAGDriver
+from ray.serve.deployment_graph import InputNode
+
+# These imports are used only for type hints:
+from typing import Dict, List
+from starlette.requests import Request
+from ray.serve.deployment_graph import ClassNode
+
+
+@serve.deployment(num_replicas=2)
+class FruitMarket:
+
+    def __init__(
+        self,
+        mango_stand: ClassNode,
+        orange_stand: ClassNode,
+        pear_stand: ClassNode,
+    ):
+        self.directory = {
+            "MANGO": mango_stand,
+            "ORANGE": orange_stand,
+            "PEAR": pear_stand,
+        }
+    
+    def check_price(self, fruit: str, amount: float) -> float:
+        if fruit not in self.directory:
+            return -1
+        else:
+            fruit_stand = self.directory[fruit]
+            return ray.get(fruit_stand.check_price.remote(amount))
+
+
+@serve.deployment(user_config={"price": 3})
+class MangoStand:
+
+    DEFAULT_PRICE = 1
+
+    def __init__(self):
+        # This default price is overwritten by the one specified in the
+        # user_config through the reconfigure() method.
+        self.price = self.DEFAULT_PRICE
+    
+    def reconfigure(self, config: Dict):
+        self.price = config.get("price", self.DEFAULT_PRICE)
+    
+    def check_price(self, amount: float) -> float:
+        return self.price * amount
+
+
+@serve.deployment(user_config={"price": 2})
+class OrangeStand:
+
+    DEFAULT_PRICE = 0.5
+
+    def __init__(self):
+        # This default price is overwritten by the one specified in the
+        # user_config through the reconfigure() method.
+        self.price = self.DEFAULT_PRICE
+    
+    def reconfigure(self, config: Dict):
+        self.price = config.get("price", self.DEFAULT_PRICE)
+    
+    def check_price(self, amount: float) -> float:
+        return self.price * amount
+
+
+@serve.deployment(user_config={"price": 4})
+class PearStand:
+
+    DEFAULT_PRICE = 0.75
+
+    def __init__(self):
+        # This default price is overwritten by the one specified in the
+        # user_config through the reconfigure() method.
+        self.price = self.DEFAULT_PRICE
+    
+    def reconfigure(self, config: Dict):
+        self.price = config.get("price", self.DEFAULT_PRICE)
+    
+    def check_price(self, amount: float) -> float:
+        return self.price * amount
+
+
+async def json_resolver(request: Request) -> List:
+    return await request.json()
```

Comment:
you can import this from serve.http_adapters

Reply:
Ah good catch– I imported it.